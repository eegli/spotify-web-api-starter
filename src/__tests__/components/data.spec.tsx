import { render } from '@testing-library/react';
import Data from '../../components/data.component';

describe('data renders correctly', () => {
  const mockData = {
    album_type: 'COMPILATION',
    href: 'https://api.spotify.com/v1/albums/3wIq5okaszSGMxYv4zlzyR',
    artists: [],
    disc_number: 1,
    duration_ms: 340965,
    explicit: false,
    external_ids: { isrc: 'GBKPL1817555' },
    external_urls: {
      spotify: 'https://open.spotify.com/track/2Vrg4Ar4v8KECMdwck1DRl',
    },
    id: '2Vrg4Ar4v8KECMdwck1DRl',
    is_local: false,
    name: 'Low - Kaskade Remix',
    popularity: 46,
    track_number: 2,
    type: 'track',
    uri: 'spotify:track:2Vrg4Ar4v8KECMdwck1DRl',
  };
  it('renders empty div on empty object prop', () => {
    const { getByTestId } = render(<Data color="alert-danger" data={{}} />);
    expect(getByTestId('div-empty')).toBeTruthy();
  });

  it('renders empty div on empty array prop', () => {
    const { getByTestId } = render(<Data color="alert-danger" data={[]} />);
    expect(getByTestId('div-empty')).toBeTruthy();
  });

  it('renders formatted json 1', () => {
    const { container, getByTestId } = render(
      <Data color="alert-danger" data={mockData} />
    );
    expect(getByTestId('json-data')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders formatted json 2', () => {
    const { container, getByTestId } = render(
      <Data color="alert-danger" data={[mockData]} />
    );
    expect(getByTestId('json-data')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
