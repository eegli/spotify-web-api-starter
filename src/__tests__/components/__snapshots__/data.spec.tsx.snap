// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`data renders correctly renders formatted json 1 1`] = `
<div
  class="alert alert-danger"
  role="alert"
>
  <pre
    data-testid="json-data"
  >
    {
  "album_type": "COMPILATION",
  "href": "https://api.spotify.com/v1/albums/3wIq5okaszSGMxYv4zlzyR",
  "artists": [],
  "disc_number": 1,
  "duration_ms": 340965,
  "explicit": false,
  "external_ids": {
    "isrc": "GBKPL1817555"
  },
  "external_urls": {
    "spotify": "https://open.spotify.com/track/2Vrg4Ar4v8KECMdwck1DRl"
  },
  "id": "2Vrg4Ar4v8KECMdwck1DRl",
  "is_local": false,
  "name": "Low - Kaskade Remix",
  "popularity": 46,
  "track_number": 2,
  "type": "track",
  "uri": "spotify:track:2Vrg4Ar4v8KECMdwck1DRl"
}
  </pre>
</div>
`;

exports[`data renders correctly renders formatted json 2 1`] = `
<div
  class="alert alert-danger"
  role="alert"
>
  <pre
    data-testid="json-data"
  >
    [
  {
    "album_type": "COMPILATION",
    "href": "https://api.spotify.com/v1/albums/3wIq5okaszSGMxYv4zlzyR",
    "artists": [],
    "disc_number": 1,
    "duration_ms": 340965,
    "explicit": false,
    "external_ids": {
      "isrc": "GBKPL1817555"
    },
    "external_urls": {
      "spotify": "https://open.spotify.com/track/2Vrg4Ar4v8KECMdwck1DRl"
    },
    "id": "2Vrg4Ar4v8KECMdwck1DRl",
    "is_local": false,
    "name": "Low - Kaskade Remix",
    "popularity": 46,
    "track_number": 2,
    "type": "track",
    "uri": "spotify:track:2Vrg4Ar4v8KECMdwck1DRl"
  }
]
  </pre>
</div>
`;
